OWML.Common/IHarmonyHelper.cs

```diff
@@ -3,15 +3,57 @@
 
 namespace OWML.Common
 {
+    /// <summary>
+    /// Handler for patching methods.
+    /// </summary>
     public interface IHarmonyHelper
     {
+        /// <summary>Add prefix patch to the given method.</summary>
+        /// <typeparam name="T">The type the method is in.</typeparam>
+        /// <param name="methodName">The name of the method.</param>
+        /// <param name="patchType">The type the patch is in.</param>
+        /// <param name="patchMethodName">The name of the patch method.</param>
```

Comment:
I think we can make the descriptions here a lot more helpful. Right now it does't seem to say much more than the parameter names already say. Not sure if it would be too much information to have here though. For instance:

1. The generic version of `AddPrefix` only works if there is only one method with the provided `methodName`;
2. The generic version also doesn't work if the provided type parameter `T` belongs to a static class;
3. If 1 or 2 are the case, we should use the non-generic version of AddPrefix, with a MethodInfo instead;
4. `patchMethodName` needs to be the name of a method in the `patchType` class;
5. The patch method needs to be static.

I think this would be enough information to make it possible to write a patch without having to read any documentation, except perhaps for knowing that HarmonyHelper.AddPrefix exists. Do you think we should keep this info for external documentation? I think it would be pretty cool to be able to understand all these restrictions right away, but might be overkill.

Reply:
Done.